dao/mongodb.rya/src/main/java/mvm/rya/mongodb/MongoDBRyaDAO.java

```diff
@@ -73,14 +73,9 @@
      */
     public MongoDBRyaDAO(final MongoDBRdfConfiguration conf) throws RyaDAOException, NumberFormatException, UnknownHostException {
         this.conf = conf;
-        try {
-            mongoClient = MongoConnectorFactory.getMongoClient(conf);
-            conf.setMongoClient(mongoClient);
-            init();
-        } catch (NumberFormatException | UnknownHostException e) {
-            log.error("Unable to create a connection to mongo.", e);
-            throw e;
-        }
+        mongoClient = MongoConnectorFactory.getMongoClient(conf);
```

Comment:
the UnknownHostException is a checked exception and must be caught.  I removed both of the exceptions and wrapped into better unchecked exceptions.  I assume that the default error handler will log the error.  Otherwise we should attempt to log every unchecked exception.  I made the underlying thrown errors more descriptive, they all mention that the MongoDB connection has failed, so this log entry is redundant.